### PR TITLE
Companies house (independent publisher)

### PR DESCRIPTION
**Pull request by MattCollins-Jones**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: independent-publisher-connectors/Companies House/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,2615 @@
...
+                    "address_line_2": {
+                      "type": "string",
+                      "description": "address_line_2"
+                    },
+                    "locality": {
+                      "type": "string",
+                      "description": "locality"
+                    }
+                  },
+                  "description": "registered_office_address"
```
Comment (sriyen-msft) [inline independent-publisher-connectors/Companies House/apiDefinition.swagger.json:95]: @MattCollins-Jones **All the description values of properties in the swagger** are expected to be free from grammatical and spell errors. Please provide valid description values.
Docs - [Define operation and parameter summaries and descriptions](https://docs.microsoft.com/connectors/custom-connectors/certification-submission#define-operation-and-parameter-summaries-and-descriptions)

File: independent-publisher-connectors/Companies House/apiProperties.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+          "tooltip": "Provide your Authorization",
+          "constraints": {
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#149CFA",
```
Comment (sriyen-msft) [inline independent-publisher-connectors/Companies House/apiProperties.json:18]: @MattCollins-Jones Please go thru the [template provided for apiProperties.json](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json) file, all the IP Connectors must have iconBrandColor as "#da3b01"

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/Companies House/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,2615 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "1.0.0",
+    "title": "Companies House",
+    "description": "Companies House Connector to search for information.",
```
[the remark this answers is not in the document]
Comment (MattCollins-Jones) [reply independent-publisher-connectors/Companies House/apiDefinition.swagger.json]: Hi @sriyen-msft I've added a longer more detailed description which I hope satisfies this requirement. If not, let me know and i'll work on this further.

File: independent-publisher-connectors/Companies House/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,2615 @@
...
+    "/company/%7B%7BCompanyNumber%7D%7D/persons-with-significant-control-statements/%7B%7BStatementId%7D%7D": {},
+    "/company/%7B%7BCompanyNumber%7D%7D/uk-establishments": {},
+    "//officers/%7B%7BOfficerId%7D%7D/appointments": {},
+    "/company/%7B%7BCompanyNumber%7D%7D/filing-history/%7B%7BTransactionId%7D%7D": {},
+    "/company/%7B%7BCompanyNumber%7D%7D/charges": {},
+    "/company/%7B%7BCompanyNumber%7D%7D/charges/%7B%7BChargeId%7D%7D": {},
+    "/company/%7B%7BCompanyNumber%7D%7D/registers": {},
+    "/company/%7B%7BCompanyNumber%7D%7D/registered-office-address": {},
+    "/company/%7B%7BCompanyNumber%7D%7D/officers": {},
+    "/company/%7B%7BCompanyNumber%7D%7D/appointments/%7B%7BAppointmentId%7D%7D": {},
```
[the remark this answers is not in the document]
Comment (MattCollins-Jones) [reply independent-publisher-connectors/Companies House/apiDefinition.swagger.json]: @sriyen-msft I think these are from importing from Postman to the connector and working on it from there. I wasn't sure what these were used for, but it looks like some old operations I've removed or edited. I've removed these now.

File: independent-publisher-connectors/Companies House/apiDefinition.swagger.json (hunk 1 above)
Comment (MattCollins-Jones) [reply independent-publisher-connectors/Companies House/apiDefinition.swagger.json]: @sriyen-msft I see the requirements for these now for not allowing certain characters. I have hopefully updated them all, apologies if I've missed any or if there are any additional details I need to add in.

File: independent-publisher-connectors/Companies House/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,2601 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "1.0.0",
+    "title": "Companies House",
+     "description": "This connectors uses the Companies House API to search and retrieve information about companies registered in the UK. Companies House is a UK government organisation that registers companies, recording information about them like their address, annual turnover and who is elected as an officer of the company. ",
```
[the remark this answers is not in the document]
Comment (MattCollins-Jones) [reply independent-publisher-connectors/Companies House/apiDefinition.swagger.json]: I've made this change @sriyen-msft 😊

**Comment by sriyen-msft** [comment]
@MattCollins-Jones We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps. Thanks.

**Comment by sriyen-msft** [comment]
[[certify-connector]]

**Review by sriyen-msft — APPROVED** [review]

**Comment by sriyen-msft** [comment]
@MattCollins-Jones Apologies for the delay in response. 

Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, Thanks.

**Pull request merged**